dotnet/src/Connectors/Connectors.Memory.AzureCosmosDBNoSQL/AzureCosmosDBNoSQLVectorStoreCollectionQueryBuilder.cs

```diff
@@ -125,6 +125,68 @@ public static QueryDefinition BuildSearchQuery<TVector, TRecord>(
         return queryDefinition;
     }
 
+    internal static QueryDefinition BuildSearchQuery(
+        VectorStoreRecordModel model,
+        string whereClause, Dictionary<string, object?> filterParameters,
+        IEnumerable<KeyValuePair<VectorStoreRecordPropertyModel, bool>> orderByProperties,
+        bool includeVectors,
+        int top,
+        int skip)
+    {
+        var tableVariableName = AzureCosmosDBNoSQLConstants.ContainerAlias;
+
+        IEnumerable<VectorStoreRecordPropertyModel> projectionProperties = model.Properties;
+        if (!includeVectors)
+        {
+            projectionProperties = projectionProperties.Where(p => p is not VectorStoreRecordVectorPropertyModel);
+        }
+
+        var fieldsArgument = projectionProperties.Select(field => $"{tableVariableName}.{field}");
+
+        var selectClauseArguments = string.Join(SelectClauseDelimiter, [.. fieldsArgument]);
+
+        // If Offset is not configured, use Top parameter instead of Limit/Offset
+        // since it's more optimized.
+        var topArgument = skip == 0 ? $"TOP {top} " : string.Empty;
+
+        var builder = new StringBuilder();
+
+        builder.AppendLine($"SELECT {topArgument}{selectClauseArguments}");
+        builder.AppendLine($"FROM {tableVariableName}");
+        builder.Append("WHERE ").AppendLine(whereClause);
+
+        bool orderByClauseNotAdded = true;
+        foreach (var pair in orderByProperties)
+        {
+            if (orderByClauseNotAdded)
+            {
+                builder.Append("ORDER BY ");
+                orderByClauseNotAdded = false;
+            }
+
+            builder.AppendFormat("{0}.{1} {2},", tableVariableName, pair.Key.StorageName, pair.Value ? "ASC" : "DESC");
+        }
+        if (!orderByClauseNotAdded)
```

Comment:
I'd generally recommend passing the orderByProperties as an IReadOnlyList rather than IEnumerable. This way you can just check whether Count > 0 instead of tracking this via an extra local variable etc.